### PR TITLE
Limit the number of events and deployments displayed in dc describer

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/9604

This PR will limit the events displayed to last 4 and also the deployments displayed in describe to last 3.

### CODE REVIEW

**Comment by mfojtik** [comment]
@kargakis PTAL

File: pkg/cmd/cli/describe/deployments.go [hunk 1]
```diff
@@ -93,19 +93,30 @@ func (d *DeploymentConfigDescriber) Describe(namespace, name string, settings kc
...
 				sort.Sort(sort.Reverse(rcutils.OverlappingControllers(sorted)))
+				counter := 0
 				for _, item := range sorted {
 					if item.Name != deploymentName && deploymentConfig.Name == deployutil.DeploymentConfigNameFor(&item) {
 						header := fmt.Sprintf("Deployment #%d", deployutil.DeploymentVersionFor(&item))
 						printDeploymentRc(&item, d.kubeClient, out, header, false)
+						counter++
+					}
+					// Only show last three deployments here
+					if counter == 2 {
```
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Add a comment that this number+1 needs to be on par with the respective number in [oc status](https://github.com/openshift/origin/blob/662753c7a7ee9efbbf3dbe1dd4f4bc324b976d74/pkg/cmd/cli/describe/projectstatus.go#L482)

File: pkg/cmd/cli/describe/deployments.go [hunk 2]
```diff
@@ -93,19 +93,30 @@ func (d *DeploymentConfigDescriber) Describe(namespace, name string, settings kc
...
+						break
 					}
 				}
 			}
 		}
 
 		// Events
 		if events, err := d.kubeClient.Events(deploymentConfig.Namespace).Search(deploymentConfig); err == nil && events != nil {
+			latestDeploymentEvents := &kapi.EventList{Items: []kapi.Event{}}
+			// Only show last 5 events here
```
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: I am not sure about 5. What if I have a deployment with 3 hooks? At the very least we must always show all the events for the latest deployment. @smarterclayton @fabianofranz any ideas on this?

Comment (mfojtik) [inline pkg/cmd/cli/describe/deployments.go]: yeah, however the events are for the DC, not for the deployment (RC) and I don't want to do string parsing on messages here :-)

Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Showing events for the previous deployment is not worse that what we are doing atm so a bigger number that accomodates all standard events for a deployment (2 for each hook -we could possibly identify how many hooks we have- one for the created deployment, one that it successfully deployed or failed - 8 or 10 in total) is acceptable. The gist is that we should be more sophisticated here in the way we choose the number of events to show.

Comment (mfojtik) [inline pkg/cmd/cli/describe/deployments.go]: @kargakis can we make the number of events that are shown configurable (DescriberSettings#NumOfEvents), so users that are interested in "older" events can use that option (and we can default to 8-10)

Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Making the number of events configurable sounds reasonable but we should have it as a separate issue. For now just bump this to 8 and open two follow-ups, one for the configurable # of events and one for making our default estimation more sophisticated.

File: pkg/cmd/cli/describe/deployments.go (hunk 1 above)
Comment (mfojtik) [inline pkg/cmd/cli/describe/deployments.go]: @kargakis I can make it a const

Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Oh, I didn't realize we are on the same package. Yes, please, make it a constant.

File: pkg/cmd/cli/describe/deployments.go (hunk 2 above)
Comment (mfojtik) [inline pkg/cmd/cli/describe/deployments.go]: @kargakis done, issue and const(s) created.

File: pkg/cmd/cli/describe/deployments.go [hunk 3]
```diff
@@ -27,6 +27,20 @@ import (
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
+const (
+	// MaxDisplayDeployments is the number of deployments to show when describing
+	// deployment configuration.
+	// TODO: Use this constant when limiting the number of deployments displayed in
```
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Plug this in `oc status` while here. Also we don't need to export any of these constants.

File: pkg/cmd/cli/describe/deployments.go [hunk 4]
```diff
@@ -27,6 +27,20 @@ import (
...
+	// oc status.
+	MaxDisplayDeployments = 2
```
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Should be three because we display three:)

File: pkg/cmd/cli/describe/deployments.go [hunk 5]
```diff
@@ -93,19 +107,28 @@ func (d *DeploymentConfigDescriber) Describe(namespace, name string, settings kc
 			if err == nil {
 				sorted := deploymentsHistory.Items
 				sort.Sort(sort.Reverse(rcutils.OverlappingControllers(sorted)))
+				counter := 0
```
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: you don't need a counter, take this from the for loop

Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: hm, actually you do need it, nevermind

File: pkg/cmd/cli/describe/deployments.go (hunk 4 above)
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Hm, this is just the old deployments, right?

Comment (mfojtik) [inline pkg/cmd/cli/describe/deployments.go]: we count from 0, but yeah, this should be 3.

**Comment by mfojtik** [comment]
@kargakis fixed the count and plumbed `oc status` as well.

File: pkg/cmd/cli/describe/deployments.go [hunk 6]
```diff
@@ -27,6 +27,20 @@ import (
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
+const (
+	// maxDisplayDeployments is the number of deployments to show when describing
+	// deployment configuration.
+	// TODO: Use this constant when limiting the number of deployments displayed in
```
Comment (kargakis) [inline pkg/cmd/cli/describe/deployments.go]: Remove the TODO

**Comment by kargakis** [comment]
One last comment, LGTM otherwise

**Comment by mfojtik** [comment]
@kargakis done. will merge on green tests

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**